cps/web.py

```diff
@@ -655,7 +664,25 @@ def edit_book(book_id):
     if request.method == 'POST':
         to_save = request.form.to_dict()
         book.title = to_save["book_title"]
-        book.authors[0].name = to_save["author_name"]
+        
+        author_id = book.authors[0].id
+        
+        is_author = db.session.query(db.Authors).filter(db.Authors.name.like('%' + to_save["author_name"].strip() + '%')).first()
```

Comment:
Why are you doing a wildcard match of the author name?
The wildcard will match to_save["author_name"]="Rolf Schmid" with an existing author "Rolf Schmidt". Thats why the book author doesn't get changed in that case as i previously commented.
It should be an equal operator here.